internal/process/deprovisioning/check_gardenercluster_deleted_step.go

```diff
@@ -0,0 +1,66 @@
+package deprovisioning
+
+import (
+	"context"
+	"time"
+
+	"github.com/kyma-project/kyma-environment-broker/internal/process/steps"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/kyma-project/kyma-environment-broker/internal"
+	"github.com/kyma-project/kyma-environment-broker/internal/process"
+	"github.com/kyma-project/kyma-environment-broker/internal/storage"
+)
+
+type CheckGardenerClusterDeletedStep struct {
+	operationManager *process.OperationManager
+	kcpClient        client.Client
+}
+
+func NewCheckGardenerClusterDeletedStep(operations storage.Operations, kcpClient client.Client) *CheckGardenerClusterDeletedStep {
+	return &CheckGardenerClusterDeletedStep{
+		operationManager: process.NewOperationManager(operations),
+		kcpClient:        kcpClient,
+	}
+}
+
+func (step *CheckGardenerClusterDeletedStep) Name() string {
+	return "Check_GardenerCluster_Deleted"
+}
+
+func (step *CheckGardenerClusterDeletedStep) Run(operation internal.Operation, logger logrus.FieldLogger) (internal.Operation, time.Duration, error) {
+	if operation.KymaResourceNamespace == "" {
+		logger.Warnf("namespace for Kyma resource not specified")
```

Comment:
Why here is Warn and below Info - the result is the same, we skip the step.

Reply:
namespace should be set before the step (somewhere in the provisioning initialisation)